engine/hid/src/glfw/hid_native.cpp

```diff
@@ -125,43 +123,54 @@ namespace dmHID
         // Update keyboard
         if (!context->m_IgnoreKeyboard)
         {
-            // TODO: Actually detect if the keyboard is present
-            context->m_KeyboardConnected = 1;
-            for (uint32_t i = 0; i < MAX_KEY_COUNT; ++i)
+            for (uint32_t k = 0; k < MAX_KEYBOARD_COUNT; ++k)
             {
-                uint32_t mask = 1;
-                mask <<= i % 32;
-                int key = glfwGetKey(i);
-                if (key == GLFW_PRESS)
-                    context->m_KeyboardPacket.m_Keys[i / 32] |= mask;
-                else
-                    context->m_KeyboardPacket.m_Keys[i / 32] &= ~mask;
+                Keyboard* keyboard = &context->m_Keyboards[k];
+                keyboard->m_Connected = 1; // TODO: Actually detect if the keyboard is present
+
+                bool pressed = false;
+                for (uint32_t i = 0; i < MAX_KEY_COUNT; ++i)
+                {
+                    uint32_t mask = 1;
+                    mask <<= i % 32;
+                    int state = glfwGetKey(i);
+                    if (state == GLFW_PRESS)
+                        keyboard->m_Packet.m_Keys[i / 32] |= mask;
+                    else
+                        keyboard->m_Packet.m_Keys[i / 32] &= ~mask;
```

Comment:
Moved the keyboard/mouse/touchdevice from single items in the context to be arrays of devices. This will open up for adding e.g. multiple mice/keyboards/touchdevices. Also, the code feels a bit cleaner imho.